packages/core/strapi/lib/services/entity-service/types/result.d.ts

```diff
@@ -2,24 +2,63 @@ import type { Attribute, Common, EntityService, Utils } from '@strapi/strapi';
 
 type Pagination = { page: number; pageSize: number; pageCount: number; total: number };
 
-export type Entity<TSchemaUID extends Common.UID.Schema> = Utils.Object.DeepPartial<
-  GetValues<TSchemaUID>
+export type Entity<
+  TSchemaUID extends Common.UID.Schema,
+  TParams extends EntityService.Params.Pick<TSchemaUID, 'fields' | 'populate'>
+> = Utils.Object.DeepPartial<
+  GetValues<
+    TSchemaUID,
+    As<ExtractFields<TSchemaUID, TParams['fields']>, Attribute.GetNonPopulatableKeys<TSchemaUID>>,
+    As<ExtractPopulate<TSchemaUID, TParams['populate']>, Attribute.GetPopulatableKeys<TSchemaUID>>
```

Comment:
```suggestion
    ExtractFields<TSchemaUID, TParams['fields']>,
    ExtractPopulate<TSchemaUID, TParams['populate']>
```

Reply:
_(this is errored for now, the suggestions made in the pr review comment combined with this change fixes the errors)_